eo-parser/src/main/java/org/eolang/parser/StrictXmir.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2024 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.parser;
+
+import com.jcabi.log.Logger;
+import com.jcabi.xml.StrictXML;
+import com.jcabi.xml.XML;
+import com.jcabi.xml.XMLDocument;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.List;
+import javax.xml.namespace.NamespaceContext;
+import org.cactoos.bytes.BytesOf;
+import org.cactoos.bytes.IoCheckedBytes;
+import org.cactoos.io.InputOf;
+import org.w3c.dom.Node;
+import org.xembly.Directives;
+import org.xembly.Xembler;
+import org.xml.sax.SAXParseException;
+
+/**
+ * XMIR that validates itself right after construction.
+ *
+ * <p>This class is supposed to be used ONLY for testing, because
```

Comment:
@yegor256 I'm a bit confused here. You mentioned in the PR description that this PR closes https://github.com/objectionary/eo/issues/3636. And here, in the javadoc you write 

```
This class is supposed to be used ONLY for testing
```

It means, I won't be able to use `StrictXmir` in `jeo-maven-plugin` production code. However, I'm using `StrictXML` during the `xmir` generation. In other words, I use it in the production code.

Reply:
@volodya-lombrozo you can try to do this, on your side: 1) create `XML` object, 2) save it on disc, 3) make `StrictXmir(xml)` and 4) either fail, or continue. If you fail, you fail. If you continue, you have a clean XML on disc.

Reply:
@yegor256 [NIT] I mean, this class might be used not only for testing. Actually it's an important part of validation in a production code. Maybe we need to remove this line from javadoc:

> This class is supposed to be used ONLY for testing

and replace it with something like 

> ATTENTION!

 what do you think?

Reply:
@volodya-lombrozo explained better in 04af5af65